modules/@apostrophecms/doc/index.js

```diff
@@ -1843,8 +1845,115 @@ module.exports = {
         return existing?.aposDocId || self.apos.util.generateId();
       },
 
-      ...require('./lib/legacy-migrations')(self),
-      ...require('./lib/migrations')(self)
+      async getAposDocId({
+        id, slug, locale
+      }) {
+        if (!id && !slug) {
+          throw self.apos.error('invalid', 'Either id or slug must be provided');
+        }
+        if (!locale) {
+          throw self.apos.error('invalid', 'Missing locale');
+        }
+
+        const req = self.apos.task.getReq({ mode: 'draft' });
```

Comment:
You need to put locale here too, and provide tests verifying that distinguishing two different locales when using `slug` and the slug is the same works correctly

Alternatively, go straight to mongo... I think that would be simplest and surest and not fail to find archived documents or other edge cases the developer wants to operate on.